showcase/app/controllers/page-components/modal.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { deepTracked } from 'ember-deep-tracked';
+
+import type { PageComponentsModalModel } from 'showcase/routes/page-components/modal';
+
+export default class PageComponentsModalController extends Controller {
+  declare model: PageComponentsModalModel;
+
+  @deepTracked modals = {
+    basicModalActive: false,
+    longModalActive: false,
+    formModalActive: false,
+    tabsModalActive: false,
+    dropdownModalActive: false,
+    superselectModalActive1: false,
+    superselectModalActive2: false,
+    superselectModalActive3: false,
+    dismissDisabledModalActive: false,
+    dropdownInitiatedModalActive: false,
+    dropdownInitiatedWithReturnedFocusModalActive: false,
+    deactivateModalOnCloseActive: false,
+    deactivateModalOnDestroyActive: false,
+    deactivateModalOnSubmitActive: false,
+    deactivateModalOnSubmitValidationError: false,
+  };
+
+  @tracked isDismissDisabled: boolean | undefined = undefined;
+
+  @action
+  activateModal(modal: keyof typeof this.modals) {
+    this.modals[modal] = true;
+
+    if (modal === 'dismissDisabledModalActive') {
+      this.isDismissDisabled = true;
+    }
+  }
+
+  @action
+  deactivateModal(modal: keyof typeof this.modals) {
```

Comment:
[question] the typing of this function is different from the equivalent in the `flyout.ts` file, why?

Reply:
my bad, I was looking at the old version 🤦‍♂️